### PR TITLE
Preserved aspect ratio and added plot_nodes_as_symbols!()

### PR DESCRIPTION
**Pull request by michael-prange**

### CODE REVIEW

File: src/plot.jl [hunk 1]
```diff
@@ -332,10 +333,36 @@ is set to `true` than `PyPlot.jl` is used
...
+    if width==nothing || height==nothing
+        # Compute aspect ratio
+        if true
+            p1 = ENU(LLA(m.bounds.min_y, m.bounds.min_x), m.bounds)
+            p2 = ENU(LLA(m.bounds.max_y, m.bounds.max_x), m.bounds)
+            w = abs(p2.east - p1.east)
+            h = abs(p2.north - p1.north)
+            aspect_ratio = h/w
+        else
+            # Why doesn't this work?
```
Comment (pszufe) [inline src/plot.jl:349]: Please clean the code.
Please provide this as a new implementation of `aspect_ratio` (line 42)
Note that aspect ratio is `w/h`, NOT `h/w` - see https://en.wikipedia.org/wiki/Aspect_ratio_(image)
Hence let's use the correct definition

File: src/plot.jl [hunk 2]
```diff
@@ -332,10 +333,36 @@ is set to `true` than `PyPlot.jl` is used
...
+            aspect_ratio = OpenStreetMapXPlot.aspect_ratio(m.bounds)
+        end
+
+        if width==nothing && height==nothing
+            width = 600
+            height = Int(round(aspect_ratio*width))
```
Comment (pszufe) [inline src/plot.jl:355]: once we correct `aspect_ratio` to match the actual definition `aspect_ratio*width` will be replaced by `width/aspect_ratio`

File: src/plot.jl [hunk 3]
```diff
@@ -332,10 +333,36 @@ is set to `true` than `PyPlot.jl` is used
...
+        elseif height==nothing
+            height = Int(round(aspect_ratio*width))
+        elseif width==nothing
+            width = Int(round(height/aspect_ratio))
+        end
```
Comment (pszufe) [inline src/plot.jl:360]: same correction for other lines

**Review by pszufe — CHANGES_REQUESTED** [review]
Thank you for this PR!
Could you also try to add one or two tests of the new code to `test/runtests.jl` - this will allow your code to be run and checked for notable errors whenever the project is build. Look for other plotting tests in that file.

**Comment by michael-prange** [comment]
No worries. I'll make these changes. 

If I make the changes on my local repository, do I just push them to GitHub? Or do I do yet another pull request?

**Comment by pszufe** [comment]
> No worries. I'll make these changes.

Thanks!!!
 
> If I make the changes on my local repository, do I just push them to GitHub? Or do I do yet another pull request?

As far as I remember you push then to your repo and they will be automatically seen in this pull request. So no need to create a new one. This makes indeed quite nice workflow.

**Comment by michael-prange** [comment]
I’ve made all the changes. The Push isn’t working. I get:



When I Fetch the changes, I don’t see any. Then when I attempt another Push, I get the same message. Is there another way for me to do this?

Michael

**Comment by pszufe** [comment]
Hi,
what message you get when trying to push?

**Comment by michael-prange** [comment]
PastedGraphic-1

**Comment by pszufe** [comment]
In order to get the remote changes you use `git pull` not `git fetch`.

I do not know what happened to your repo, but worst case (non expert solution) you can clone it to another directory (from your fork), copy paste the changed files and than push.

**Comment by michael-prange** [comment]
I deleted my fork and created another one with the changes. I’ve created another pull request that seemed to go without issue.

Michael

**Comment by pszufe** [comment]
I close this one since we merged the other PR

**Pull request closed without merging**